core/src/main/kotlin/endpoints/AgentManager.kt

```diff
@@ -410,29 +414,27 @@ class AgentManager(override val kodein: Kodein) : KodeinAware {
         setPackagesPrefixes(prefixes)
     }
 
-    private suspend fun AgentInfo.sync() {
+    private suspend fun AgentInfo.sync(instanceId: String, needClassSending: Boolean = false) {
         if (isRegistered) {
-            val instanceIds = instanceIds(id, buildVersion)
-            if (instanceIds.any()) {
-                logger.debug { "Agent($id, $buildVersion): starting sync for instances ${instanceIds.keys}..." }
-                val info = this
-                val duration = measureTime {
-                    instanceIds.forEach { (instanceId, _) ->
-                        wrapInstanceBusy(instanceId) {
-                            val settings = adminData(id).settings
-                            configurePackages(settings.packages)
-                            sendPlugins(info)
-                            updateSessionHeader(settings.sessionIdHeaderName)
-                            triggerClassesSending()
-                            enableAllPlugins(id)
-                        }
+            logger.debug { "Agent($id, $buildVersion): starting sync for instance $instanceId..." }
+            val info = this
+            val instanceKey = InstanceKey(AgentKey(info.id, info.buildVersion), instanceId)
+            val duration = measureTime {
+                wrapInstanceBusy(instanceKey) {
+                    val settings = adminData(id).settings
+                    configurePackages(settings.packages)
+                    sendPlugins(info)
+                    if (needClassSending) {
+                        updateSessionHeader(settings.sessionIdHeaderName)
+                        triggerClassesSending()
                     }
+                    enableAllPlugins(id)
                 }
-                logger.info { "Agent $id: sync took: $duration." }
-                topicResolver.sendToAllSubscribed(WsRoutes.AgentBuilds(id))
-                logger.debug { "Agent $id: sync finished." }
-            } else logger.error { "Agent $id: no instances to sync!" }
-        } else logger.warn { "Agent $id: cannot sync, status is ${AgentStatus.NOT_REGISTERED}." }
+            }
+            logger.info { "Agent $id, instance $instanceId: sync took: $duration." }
```

Comment:
let's add buildVersion to every log:
```suggestion
            logger.info { "Agent($id, $buildVersion), instance $instanceId: sync took: $duration." }
```